helpers/retryablehttp/logger.go

```diff
@@ -0,0 +1,30 @@
+package retryablehttp
+
+import (
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/rs/zerolog"
+)
+
+type leveledLogger struct {
+	zerolog.Logger
+}
+
+func (l *leveledLogger) Debug(msg string, keysAndValues ...any) {
+	l.Logger.Debug().Fields(keysAndValues).Msg(msg)
+}
+
+func (l *leveledLogger) Error(msg string, keysAndValues ...any) {
+	l.Logger.Error().Fields(keysAndValues).Msg(msg)
+}
+
+func (l *leveledLogger) Info(msg string, keysAndValues ...any) {
+	l.Logger.Info().Fields(keysAndValues).Msg(msg)
+}
+
+func (l *leveledLogger) Warn(msg string, keysAndValues ...any) {
+	l.Logger.Warn().Fields(keysAndValues).Msg(msg)
+}
+
+func NewLeveledLogger(logger zerolog.Logger) retryablehttp.LeveledLogger {
```

Comment:
You still have to instantiate this everywhere that you use the retryable, right?

Reply:
Yes, I initially wrapped `retryablehttp` and did `NewRetryableClient` but then realized I would have to add all the different options as well (as some plugins modify them), so I think sharing only the logger is better

Reply:
We have something similar in https://github.com/cloudquery/plugin-sdk/blob/5c77cee87d9fca292e9e81663c9ce3775962a623/helpers/grpczerolog/grpc.go#L12